src/ArtifactUtils.jl

```diff
@@ -70,4 +78,152 @@ function add_artifact!(
     return git_tree_sha1
 end
 
+"""
+    artifact_from_directory(source) -> artifact_id::SHA1
+
+Create an artifact from the `source` directory and return the `artifact_id`.
+"""
+artifact_from_directory(source) =
+    create_artifact() do artifact_dir
+        @show readdir(source)
+        @show read.(readdir(source; join = true), String) 
+        @show SHA1(GitTools.tree_hash(source))
+        cp(source, artifact_dir; force = true, follow_symlinks = true)
+        @show readdir(artifact_dir)
+        @show read.(readdir(artifact_dir; join = true), String) 
+        @show SHA1(GitTools.tree_hash(artifact_dir))
```

Comment:
This is printing

```
readdir(source) = ["hello.txt"]
read.(readdir(source; join = true), String) = ["Hello, world.\n"]
SHA1(GitTools.tree_hash(source)) = SHA1("0a890bd10328d68f6d85efd2535e3a4c588ee8e6")
readdir(artifact_dir) = ["hello.txt"]
read.(readdir(artifact_dir; join = true), String) = ["Hello, world.\n"]
SHA1(GitTools.tree_hash(artifact_dir)) = SHA1("952cfce0fb589c02736482fa75f9f9bb492242f8")
```

in Windows https://github.com/simeonschaub/ArtifactUtils.jl/runs/4495523331?check_suite_focus=true#step:6:91

This example is taken from Pkg's test https://github.com/JuliaLang/Pkg.jl/blob/89286eac216164c43cc996f1f31a9fa1f1dacf87/test/new.jl#L2553-L2572

`SHA1(GitTools.tree_hash(artifact_dir)) = SHA1("952cfce0fb589c02736482fa75f9f9bb492242f8")` matches with [the example with the executable bit set](https://github.com/JuliaLang/Pkg.jl/blob/89286eac216164c43cc996f1f31a9fa1f1dacf87/test/new.jl#L2571). So, I think this is due to `cp` not preserving the permission (and the executable bit is set by default in Windows?).

Reply:
I wonder if it makes sense to add `preserve` option to `Base.cp`? It can be backported through Compat.jl and we can use it here. Not sure how much it can be made cross-platform though.

The easiest solution here is probably just noting this behavior for Windows.

Reply:
Hmm, that's weird... @giordano Do you have any ideas? Otherwise we can also just mark this functionality as broken on Windows for the time being.

Reply:
Ohh Windows... No, I don't have have a good solution at the moment.

Reply:
Maybe have a look at https://github.com/JuliaIO/Tar.jl/blob/6a946029685639b69ce5a7cc4c4a6c0e6c6b2697/src/extract.jl#L143, it looks related

Reply:
Uhm, after reading the pull request which introduced that, I'm not entirely sure it's actually related

Reply:
Thanks! The code you quoted gave me the illusion that supporting this could actually be easier than noting this as a known bug.

Anyway, it turned out Pkg has implemented quite some workarounds for Windows that we need to replicate here. But it looks like 6099eff60be19d4e7c679518f62f205f3f1e0874 and 8d005d2e1d309529140322c3d7951f3c8ec9ac99 fixed the issue for Windows